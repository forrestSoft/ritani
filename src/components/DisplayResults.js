import React from 'react';

function DisplayResults(){
	return(
		<React.Fragment>
			<p>I am results</p>
		</React.Fragment>
	)
}

export default DisplayResults;