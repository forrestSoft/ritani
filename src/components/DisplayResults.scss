.display_results{
	padding: 3.5vw;

	div{
		display:flex;
		flex-wrap: wrap;
		justify-content: center;
	}

	button{
		font-size: 2.1vw;
		padding: 1vw 10vw;
		margin-top: 3vw;
		cursor: pointer;
	}

	p{
    font-size: 2.5vw;
    font-weight: 800;
    margin-top: 0;
	}
}