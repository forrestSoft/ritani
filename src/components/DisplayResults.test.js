import React from 'react';
import {shallow} from 'enzyme';
import shallowToJson from 'enzyme-to-json'
import DisplayResults from './DisplayResults';
import staticState from '../staticState'

it('render matches snapshot', () => {
	const displayResults0 = shallowToJson(shallow(<DisplayResults selectedUser={'bob'} totalFollowers={3} users={staticState.slice(0,3)} />))
  expect(displayResults0).toMatchSnapshot();

  const displayResults1 = shallowToJson(shallow(<DisplayResults selectedUser={'bob'} totalFollowers={3} />))
  expect(displayResults1).toMatchSnapshot();

  const displayResults2 = shallowToJson(shallow(<DisplayResults selectedUser={'bob'}  />))
  expect(displayResults2).toMatchSnapshot();

  const displayResults3 = shallowToJson(shallow(<DisplayResults />))
  expect(displayResults3).toMatchSnapshot();
});

it('handles positive Moar button visibility', () => {
	const displayResults = shallow(<DisplayResults hasMoar={true} selectedUser={'bob'}/>)
	expect(displayResults.find('button').length).toBeTruthy()
})

it('handles negative Moar button visibility', () => {
	const displayResults = shallow(<DisplayResults hasMoar={false}/>)
	expect(displayResults.find('button').length).toBeFalsy()
})

it('calls passed handleClick on "onClick"', () => {
	const mock = jest.fn().mockReturnValue('default')
	const displayResults = shallow(<DisplayResults handleClick={mock} hasMoar={true} selectedUser={'bob'} />)
	displayResults.find('button').simulate('click')
	expect(mock).toBeCalled()
})