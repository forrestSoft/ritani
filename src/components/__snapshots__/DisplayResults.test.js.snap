// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<section
  className="display_results"
>
  <h2>
    <em>
      bob
    </em>
     has 
    3
     followers
  </h2>
  <div>
    <Avatar
      key="https://avatars2.githubusercontent.com/u/1555350?v=4"
      url="https://avatars2.githubusercontent.com/u/1555350?v=4"
      username="myhduck"
    />
    <Avatar
      key="https://avatars0.githubusercontent.com/u/5945?v=4"
      url="https://avatars0.githubusercontent.com/u/5945?v=4"
      username="trevor"
    />
    <Avatar
      key="https://avatars0.githubusercontent.com/u/8826?v=4"
      url="https://avatars0.githubusercontent.com/u/8826?v=4"
      username="ogijun"
    />
  </div>
</section>
`;

exports[`render matches snapshot 2`] = `
<section
  className="display_results"
>
  <h2>
    <em>
      bob
    </em>
     has 
    3
     followers
  </h2>
  <div />
</section>
`;

exports[`render matches snapshot 3`] = `
<section
  className="display_results"
>
  <h2>
    <em>
      bob
    </em>
     has 
     followers
  </h2>
  <div />
</section>
`;

exports[`render matches snapshot 4`] = `
<Fragment>
  <h2>
    Welcome. Please search to the left
  </h2>
  <p>
    The code for this project is here: 
    <a
      href="https://github.com/forrestSoft/ritani"
      target="blank"
    >
      GitHub â‡›
    </a>
  </p>
</Fragment>
`;
